modules/volunteers/admin/sitesupport/editHours.js

```diff
@@ -0,0 +1,116 @@
+/* jshint esversion: 6 */
+/* globals apiRequest, confirmbox, showSidebar, simpleObjectToRequest */
+
+export default {
+  data() {
+    return {
+      record: {
+        hours: 0,
+        member: {
+          first_name: '',
+        },
+        enterer: {
+          id: '',
+        },
+        authorizer: {
+          id: '',
+        },
+        department: {
+          name: '',
+        },
+        end: null,
+      },
+    }
+  },
+  methods: {
+    commitHours() {
```

Comment:
Not sure if we're able to do so within the components, but I think we can - it might be nice to prefer `async / await` here rather than `promise.then()` syntax. It's a bit cleaner and allows us to do `try/catch` instead of `promise.then().catch()` and can be a bit cleaner code wise.

An example of how this would look is like this

```
methods: {
  async commitHours() {
     // Code before the API request

    await apiRequest('PUT', '/volunteer/hours/' + this.record.id + '?force=1', simpleObjectToRequest(item));
    location.reload();
  }
}
```

In more complex scenarios you could assign the result of the request to a variable like anything else:

`const response = await apiRequest()`

Reply:
Could definitely be a "as we clean up the code" thing - no rush on this of course.

Reply:
I am unfamiliar with async/await so I would love to learn more about them.  I say we stick with promise then for now but if there is a good way to  open up async/await as a option that would be great!